src/libbson/src/bson/bson.h

```diff
@@ -517,6 +517,10 @@ bson_as_json_with_opts (const bson_t *bson,
                         size_t *length,
                         const bson_json_opts_t *opts);
 
+BSON_EXPORT (void)
+bson_json_opts_set_outermost_array (bson_json_opts_t *opts,
```

Comment:
Add a new .rst page for `bson_json_opts_set_outermost_array`. Link the new page in the table-of-contents of [bson_json_opts_t.rst](https://github.com/mongodb/mongo-c-driver/blob/ce7468863b9a9ef13d155f0d6f1517adf011aa66/src/libbson/doc/bson_json_opts_t.rst#L50)

Comment similarly applies to `bson_array_as_relaxed_extended_json` and `bson_array_as_canonical_extended_json`.

https://github.com/mongodb/mongo-c-driver/blob/master/CONTRIBUTING.md#documentation describes how to build documentation:

> To build the documentation, it is recommended to use the Poetry-managed Python
> project to ensure that the exact tooling versions match. If you do not have
> Poetry installed, you can use the `poetry.sh` or `poetry.ps1` scripts in the
> `tools/` directory. First, install dependencies:
> 
> ```sh
> ./tools/poetry.sh install --with=docs
> ```
> 
> Then, execute `sphinx-build` in the Python environment, using the paths to the
> documentation to be generated:
> 
> ```sh
> ./tools/poetry.sh run sphinx-build -WEn -bhtml src/libmongoc/doc/ src/libmongoc/doc/html
> ```